tracer/src/Datadog.Trace/Tracer.cs

```diff
@@ -443,12 +444,24 @@ internal SpanContext CreateSpanContext(ISpanContext parent = null, string servic
                 traceContext.SetSamplingPriority(samplingPriority);
 
                 if (traceId == TraceId.Zero &&
-                    Activity.ActivityListener.GetCurrentActivity() is Activity.DuckTypes.IW3CActivity { TraceId: { } activityTraceId })
+                    Activity.ActivityListener.GetCurrentActivity() is Activity.DuckTypes.IW3CActivity { TraceId: { } activityTraceId } activity)
                 {
-                    // if there's an existing Activity we try to use its TraceId,
-                    // but if Activity.IdFormat is not ActivityIdFormat.W3C, it may be null or unparsable
-                    rawTraceId = activityTraceId;
-                    HexString.TryParseTraceId(activityTraceId, out traceId);
+                    bool useActivityTraceId = true;
+
+                    // Handlers[0] is the IgnoreActivityHandler - if it _should_ listen to an activity, that activity _should_ be ignored
+                    if (activity is Activity.DuckTypes.IActivity5 activity5 && ActivityHandlersRegister.Handlers[0].ShouldListenTo(activity5.Source.Name, activity5.Source.Version))
```

Comment:
IMO we could/should expose the `IgnoreActivityHandler` directly on `ActivityHandlersRegister` to avoid "guessing" that it's the first one? e.g

```csharp
internal static class ActivityHandlersRegister
{
    // Ignore Activity Handler catches existing integrations that also emits activities.
    internal static readonly IgnoreActivityHandler IgnoreHandler = new();

    // Activity handlers in order, the first handler where ShouldListenTo returns true will always handle that source.
    internal static readonly IActivityHandler[] Handlers =
    {
        IgnoreActivityHandler,

        // Azure Service Bus handlers
        new AzureServiceBusActivityHandler(),

        // The default handler catches an activity and creates a datadog span from it.
        new DefaultActivityHandler(),
```

Reply:
I like that idea